### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by Tinsane**

### CODE REVIEW

File: internal/backup_list_handler.go [hunk 1]
```diff
@@ -107,26 +107,12 @@ func WritePrettyBackupListDetails(backupDetails []BackupDetail, output io.Writer
...
-	var _ error
 	if pretty {
-		bytes, _ = json.MarshalIndent(backupDetails, "", "    ")
+		bytes, _ = json.MarshalIndent(data, "", "    ")
 	} else {
-		bytes, _ = json.Marshal(backupDetails)
+		bytes, _ = json.Marshal(data)
 	}
-	output.Write(bytes)
+	_, _ = output.Write(bytes)
```
Comment (g0djan) [inline internal/backup_list_handler.go]: Unhandled error

File: internal/databases/mysql/stream_fetch_handler.go [hunk 2]
```diff
@@ -1,47 +1,37 @@
...
+}
+
+func (settings BinlogFetchSettings) GetLogFolderPath() string {
+	return BinlogPath
+}
 
-	tracelog.InfoLogger.Println("stream-fetch")
-	streamSentinel, err := backup.FetchStreamSentinel()
+func (settings BinlogFetchSettings) GetFilePath(dstFolder string, logName string) (string, error) {
+	return path.Join(dstFolder, logName), nil
```
Comment (g0djan) [inline internal/databases/mysql/stream_fetch_handler.go]: Why you ensure that dir exists in oplog.GetFilePath and don't do it here?

File: internal/stream_fetch_helper.go [hunk 3]
```diff
@@ -67,46 +81,76 @@ func DownloadAndDecompressStream(backup *Backup) error {
...
 }
 
-// DownloadOplogFiles downloads files to specified folder
-func DownloadOplogFiles(oplogFiles []storage.Object, oplogFolder storage.Folder, oplogDstFolder string, logFileName string) error {
-	for _, oplogFile := range oplogFiles {
-		oplogName := utility.TrimFileExtension(oplogFile.GetName())
-		oplogFileSubFolder := path.Join(oplogDstFolder, oplogName)
-		_, err := NewDiskDataFolder(oplogFileSubFolder)
+// DownloadLogFiles downloads files to specified folder
+func DownloadLogFiles(logFiles []storage.Object, logFolder storage.Folder, logDstFolder string, getLogFilePath func(string, string) (string, error)) error {
```
Comment (g0djan) [inline internal/stream_fetch_helper.go]: Is it possible to simplify it :/ ?

File: internal/databases/mysql/stream_fetch_handler.go (hunk 2 above)
Comment (Tinsane) [reply internal/databases/mysql/stream_fetch_handler.go]: I'm not just ensuring the existence of the log dir, but creating it there.

Oplogs lie in such a way:
logFolder/*oplogName*/oplog.bson

Binlogs lie in the other way:
logFolder/*binlogName*

File: internal/stream_fetch_helper.go (hunk 3 above)
Comment (Tinsane) [reply internal/stream_fetch_helper.go]: It is definitely possible. But it appears to be a complicated task, so I decided to leave it as it is.

**Pull request merged**